eZ/Publish/Core/FieldType/Image/IO/Legacy.php

```diff
@@ -149,7 +150,13 @@ public function loadBinaryFile($binaryFileId)
     public function loadBinaryFileByUri($binaryFileUri)
     {
         try {
-            return $this->publishedIOService->loadBinaryFileByUri($binaryFileUri);
+            $image = $this->publishedIOService->loadBinaryFileByUri($binaryFileUri);
+
+            if ($image instanceof MissingBinaryFile) {
+                throw new InvalidArgumentException('binaryFileUri', sprintf("Can't find file with url {0}", $binaryFileUri));
```

Comment:
Huh, learn something every day :) It looks needlessly convoluted to me, but presumably Symfony has a good reason. Translation would be a good reason - we're not doing that here, but granted that would be easier to do if the string is already using sprintf.